gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -791,6 +791,8 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
       out_mem[i] = gst_allocator_alloc (NULL, out_tensors[i].size, NULL);
       g_assert (gst_memory_map (out_mem[i], &out_info[i], GST_MAP_WRITE));
 
+      /* init memory block */
+      memset (out_info[i].data, 0, out_info[i].size);
```

Comment:
I do not understand the need of this.

The filter is supposed to output data of size `out_info[i].size`.
In this case, the `out_info[i].data` is completely filled by the filter, and setting `out_info[i].data` this 0 will be redundant.

However, if the filter does not fill `out_info[i].data` completely, then the `out_info[i].size` is set wrong (bigger than the size of the output from filter). I am assuming that the output of the filter is not sparse.

Please correct me if I am wrong.

Reply:
Related  pr #1259
In some  case, uninitialized mem may cause error

Reply:
Related  pr #1259
In some  case, uninitialized mem may cause error

Reply:
I thought it is better to init memory in filter-main.

Reply:
actually, the filter subplugin is supposed to fill in the whole output buffer. If it is not filled, it's a bug of that subplugin. I don't think this is necessary.